OCPBUGS#30993: Doc improvements related to RHACM procs

sr1kar99 opened the pull request:
<!--- PR title format: [GH#<gh-issue-id>][BZ#<bz-issue-id>][OCPBUGS#<jira-issue-id>][OSDOCS#<jira-issue-id>]: <short-description-of-the-pr> --->

<!--- If your changes apply to the latest release and/or in-development version of OpenShift, open your PR against the `main` branch.

* For more details about the information requested in this template, see:
  https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/create_or_edit_content.adoc#submit-PR --->

Version(s):
4.15+
<!--- Specify the version or versions of OpenShift your PR applies to. -->

Issue:
[OCPBUGS-30993](https://issues.redhat.com/browse/OCPBUGS-30993)
<!--- Add a link to the Bugzilla, Jira, or GitHub issue, if applicable. --->

Link to docs preview:
[Preview](https://73279--ocpdocs-pr.netlify.app/openshift-enterprise/latest/storage/persistent_storage/persistent_storage_local/persistent-storage-using-lvms#lvms-installing-odf-logical-volume-manager-operator-using-rhacm_logical-volume-manager-storage)
<!--- Add direct link(s) to the exact page(s) with updated content from the preview build. --->

QE review:
- [x] QE has approved this change.
<!--- QE approval is required to merge a PR except for changes that do not impact the meaning of the docs. --->

Additional information:
<!--- Optional: Include additional context or expand the description here.--->

<!--- After you open your PR, ask for review from the OpenShift docs team:
  For community authors: Tag @openshift/team-documentation in a GitHub comment.--->

sr1kar99 commented:
@radeore 
Can you please review this PR?
Thanks.

radeore commented on modules/lvms-creating-lvmcluster-using-rhacm.adoc:
```diff
@@ -0,0 +1,75 @@
+[source, yaml]
+----
+apiVersion: policy.open-cluster-management.io/v1
+      kind: ConfigurationPolicy
+      metadata:
+        name: lvms
+      spec:
+        object-templates:
+           - complianceType: musthave
+             objectDefinition:
+               apiVersion: lvm.topolvm.io/v1alpha1
+               kind: LVMCluster
+               metadata:
+                 name: my-lvmcluster
+                 namespace: openshift-storage
+               spec:
+                 storage:
+                   deviceClasses: <1>
+                   - name: vg1
+                     default: true
+                     deviceSelector: <2>
+                       paths:
+                       - /dev/disk/by-path/pci-0000:87:00.0-nvme-1
+                       - /dev/disk/by-path/pci-0000:88:00.0-nvme-1
+                       optionalPaths:
+                       - /dev/disk/by-path/pci-0000:89:00.0-nvme-1
+                       - /dev/disk/by-path/pci-0000:90:00.0-nvme-1
+                     thinPoolConfig: <3>
+                       name: thin-pool-1
+                       sizePercent: 90
+                       overprovisionRatio: 10
+                     nodeSelector: <4>
+                       nodeSelectorTerms:
+                       - matchExpressions:
+                           - key: app
+                             operator: In
+                             values:
+                             - test1
+        remediationAction: enforce
+        severity: low
```
apiVersion: policy.open-cluster-management.io/v1
kind: ConfigurationPolicy
metadata:
  name: lvms
spec:
  object-templates:
  - complianceType: musthave
    objectDefinition:
      apiVersion: lvm.topolvm.io/v1alpha1
      kind: LVMCluster
      metadata:
        name: my-lvmcluster
        namespace: openshift-storage
      spec:
        storage:
          deviceClasses: <1>
          - name: vg1
            default: true
            deviceSelector: <2>
              paths:
              - /dev/disk/by-path/pci-0000:87:00.0-nvme-1
              - /dev/disk/by-path/pci-0000:88:00.0-nvme-1
              optionalPaths:
              - /dev/disk/by-path/pci-0000:89:00.0-nvme-1
              - /dev/disk/by-path/pci-0000:90:00.0-nvme-1
            thinPoolConfig: <3>
              name: thin-pool-1
              sizePercent: 90
              overprovisionRatio: 10
            nodeSelector: <4>
              nodeSelectorTerms:
              - matchExpressions:
                  - key: app
                    operator: In
                    values:
                    - test1
  remediationAction: enforce
  severity: low

radeore commented on modules/lvms-creating-lvmcluster-using-rhacm.adoc:
```diff
@@ -0,0 +1,75 @@
+[source, yaml]
+----
+apiVersion: policy.open-cluster-management.io/v1
+      kind: ConfigurationPolicy
+      metadata:
+        name: lvms
+      spec:
+        object-templates:
+           - complianceType: musthave
+             objectDefinition:
+               apiVersion: lvm.topolvm.io/v1alpha1
+               kind: LVMCluster
+               metadata:
+                 name: my-lvmcluster
+                 namespace: openshift-storage
+               spec:
+                 storage:
+                   deviceClasses: <1>
+                   - name: vg1
+                     default: true
+                     deviceSelector: <2>
+                       paths:
+                       - /dev/disk/by-path/pci-0000:87:00.0-nvme-1
+                       - /dev/disk/by-path/pci-0000:88:00.0-nvme-1
+                       optionalPaths:
+                       - /dev/disk/by-path/pci-0000:89:00.0-nvme-1
+                       - /dev/disk/by-path/pci-0000:90:00.0-nvme-1
+                     thinPoolConfig: <3>
+                       name: thin-pool-1
+                       sizePercent: 90
+                       overprovisionRatio: 10
+                     nodeSelector: <4>
+                       nodeSelectorTerms:
+                       - matchExpressions:
+                           - key: app
+                             operator: In
+                             values:
+                             - test1
+        remediationAction: enforce
+        severity: low
```
```suggestion
apiVersion: policy.open-cluster-management.io/v1
kind: ConfigurationPolicy
metadata:
  name: lvms
spec:
  object-templates:
  - complianceType: musthave
    objectDefinition:
      apiVersion: lvm.topolvm.io/v1alpha1
      kind: LVMCluster
      metadata:
        name: my-lvmcluster
        namespace: openshift-storage
      spec:
        storage:
          deviceClasses: <1>
          - name: vg1
            default: true
            deviceSelector: <2>
              paths:
              - /dev/disk/by-path/pci-0000:87:00.0-nvme-1
              - /dev/disk/by-path/pci-0000:88:00.0-nvme-1
              optionalPaths:
              - /dev/disk/by-path/pci-0000:89:00.0-nvme-1
              - /dev/disk/by-path/pci-0000:90:00.0-nvme-1
            thinPoolConfig: <3>
              name: thin-pool-1
              sizePercent: 90
              overprovisionRatio: 10
            nodeSelector: <4>
              nodeSelectorTerms:
              - matchExpressions:
                  - key: app
                    operator: In
                    values:
                    - test1
  remediationAction: enforce
  severity: low
```

sr1kar99 replied:
Updated as per the suggestion.

radeore commented on modules/lvms-creating-lvmcluster-using-rhacm.adoc:
```diff
@@ -0,0 +1,75 @@
++
+[source,terminal]
+----
+# oc create -f <file_name> -n <namespace>
```
```suggestion
# oc create -f <file_name> -n <ocp-cluster-namespace>
```

radeore replied:
`ocp-cluster-namespace` is the namespace of the ocp-cluster which was selected to install lvms operator

sr1kar99 replied:
Updated as per the suggestion.

radeore commented on modules/lvms-deleting-lvmcluster-using-rhacm.adoc:
```diff
@@ -0,0 +1,184 @@
+.Procedure
+
+. Log in to the {rh-rhacm} CLI using your {product-title} credentials.
+. Create a `Policy` CR YAML file to delete the `LVMCluster` CR:
```
```suggestion
. Delete ConfigurationPolicy which was created for LVMCluster CR
reate the `LVMCluster` CR by running the following command:
+
[source,terminal]
----
# oc delete -f <file_name> -n <ocp-cluster-namespace>
----
. Create a `Policy` CR YAML file to delete the `LVMCluster` CR:
```

sr1kar99 replied:
Updated as per the suggestion.

radeore commented:
@sr1kar99 Verified the procedures on ACM cluster, suggested few changes, PTAL. Thanks!

radeore commented:
/lgtm

sr1kar99 commented:
/label peer-review-needed

skrthomas commented on modules/lvms-deleting-lvmcluster-using-rhacm.adoc:
```diff
@@ -0,0 +1,192 @@
+
+.Verification
+
+. Check the status of the `Policy` CRs by running the following command:
```
```suggestion
* Check the status of the `Policy` CRs by running the following command:
```

skrthomas replied:
All your others are fine, but just this one needs to have a bullet instead of the single step https://redhat-documentation.github.io/modular-docs/#con-creating-procedure-modules_writing-mod-docs

sr1kar99 replied:
Updated as per the suggestion.

skrthomas commented on modules/lvms-installing-logical-volume-manager-operator-using-rhacm.adoc:
```diff
@@ -112,73 +116,21 @@ spec:
---
+<1> Set the `key` field and `values` field in `PlacementRule.spec.clusterSelector` to match the labels that are configured in the clusters on which you want to install {lvms}.
+<2> Namespace configuration.
+<3> `OperatorGroup` CR configuration.
+<4> `Subscription` CR configuration.
```
I would suggest placing a "The" before  the specs in <3> and <4>.

sr1kar99 replied:
Updated as per the suggestion.

skrthomas commented on modules/lvms-scaling-storage-of-clusters-using-rhacm.adoc:
```diff
@@ -0,0 +1,70 @@
+                       optionalPaths: <3>
+                       - /dev/disk/by-path/pci-0000:89:00.0-nvme-1
+# ...
+
```
remove this space at line 56

sr1kar99 replied:
Updated as per the suggestion.

skrthomas commented on modules/lvms-scaling-storage-of-clusters-using-rhacm.adoc:
```diff
@@ -0,0 +1,70 @@
+
+. In the `LVMCluster` CR, add the path to the new device in the `deviceSelector` field.
++
+.Example `LVMCluster` CR
```
```suggestion
.Example `LVMCluster` CR:
```

skrthomas replied:
Add a colon here to introduce the example CR

sr1kar99 replied:
Updated as per the suggestion.

skrthomas reviewed (commented):
A few comments for you. Nice work :) Its all going to come together nicely!

sr1kar99 commented:
@jakobmoellerdev 
Can you please review this PR?
Thanks.

sr1kar99 commented on modules/lvms-creating-lvmcluster-using-rhacm.adoc:
```diff
@@ -0,0 +1,76 @@
+
+. Log in to the {rh-rhacm} CLI using your {product-title} credentials.
+
+. Create an `LVMCluster` CR YAML file:
```
Can we update it as follows:

Create a `ConfigurationPolicy` CR YAML file with the configuration to create an `LVMCluster` CR.

sr1kar99 commented on modules/lvms-creating-lvmcluster-using-rhacm.adoc:
```diff
@@ -0,0 +1,76 @@
+<3> Contains the LVM thin pool configuration.
+<4> Contains the configuration to choose the nodes on which you want to create the LVM volume groups. If this field is empty, then all nodes without no-schedule taints are considered.
+
+. Create the `LVMCluster` CR by running the following command:
```
Updated as follows:

Create the `ConfigurationPolicy` CR by running the following command:

sr1kar99 commented on modules/lvms-creating-lvmcluster-using-rhacm.adoc:
```diff
@@ -0,0 +1,76 @@
+spec:
+  object-templates:
+  - complianceType: musthave
+    objectDefinition:
```
This `ConfigurationPolicy` CR example has few parameters that are not included in the `LVMCluster` CR example:

For example:

apiVersion: policy.open-cluster-management.io/v1
kind: ConfigurationPolicy
metadata:
  name: lvms
spec:
  object-templates:
  - complianceType: musthave
  
  
Updated the example as follows:

----
apiVersion: policy.open-cluster-management.io/v1
kind: ConfigurationPolicy
metadata:
  name: lvms
spec:
  object-templates:
  - complianceType: musthave
    objectDefinition:
      apiVersion: lvm.topolvm.io/v1alpha1
      kind: LVMCluster
      metadata:
        name: my-lvmcluster
        namespace: openshift-storage
      spec:
        storage:
          deviceClasses: <1>
# ...
            deviceSelector: <2>
# ...
            thinPoolConfig: <3>
# ...
            nodeSelector: <4>
# ...
  remediationAction: enforce
  severity: low
----

Provided a brief desc of a few parameters. The following has been added as a pre-req in all the "Create LVMCluster" CR procs:

" You read the "About the LVMCluster custom resource" section."

sr1kar99 commented on modules/lvms-deleting-lvmcluster-using-rhacm.adoc:
```diff
@@ -0,0 +1,192 @@
+
+.Prerequisites
+
+* You have access to the {rh-rhacm} cluster as a user with `cluster-admin` permissions.
```
@radeore 
Could you please confirm?

radeore replied:
I verified all procedures  as 'cluster-admin'. But not sure if it's mandatory.

sr1kar99 replied:
@radeore 
Could you please verify which all procedures need cluster-admin permission?
If it is not mandatory then it is important that we remove this point from the pre-req.

sr1kar99 replied:
> @radeore Could you please verify which all procedures need cluster-admin permission? If it is not mandatory then it is important that we remove this point from the pre-req.

Hi @radeore 
Any update on this? :)

radeore replied:
Hi @sr1kar99 , I'm checking with ACM team regarding this, will update here once I get some info from them.

radeore replied:
@sr1kar99 Checked with ACM QE, cluster-admin permissions are required for all these operations.

sr1kar99 replied:
Thanks for the confirmation, Rahul!

sr1kar99 commented on modules/lvms-scaling-storage-of-clusters-using-rhacm.adoc:
```diff
@@ -0,0 +1,69 @@
+                       - /dev/disk/by-path/pci-0000:89:00.0-nvme-1
+# ...
+----
+<1> Contains the configuration to specify the paths to the devices that you want to add to the LVM volume group.
```
As the `ConfigurationPolicy` CR had few attributes that were not present in the LVMCluster CR, I could not reference/re-use the file with LVMCluster definition in this section.

sr1kar99 commented on modules/lvms-deleting-lvmcluster-using-rhacm.adoc:
```diff
@@ -0,0 +1,192 @@
+metadata:
+  name: policy-lvmcluster-inform
+  annotations:
+    policy.open-cluster-management.io/standards: NIST SP 800-53
```
@radeore 
Could you PTAL?
Thanks.

radeore replied:
I don't have much details on why this specific standard is set, but most of the policies fall under the NIST or CPI security standards.

sr1kar99 commented:
/retest

sr1kar99 commented:
Hi @radeore 
Jakob has left a few queries. Could you PTAL?
Thanls.

sr1kar99 commented:
Hi @jakobmoellerdev 
Could you please confirm if the updates are fine?
Thanks.

jakobmoellerdev reviewed (commented):
LGTM

sr1kar99 commented:
/label merge-review-needed

adellape commented on modules/lvms-scaling-storage-of-clusters-using-rhacm.adoc:
```diff
@@ -0,0 +1,69 @@
+<1> Contains the configuration to specify the paths to the devices that you want to add to the LVM volume group.
+You can specify the device paths in the `paths` field, the `optionalPaths` field, or both. If you do not specify the device paths in both `paths` and `optionalPaths`, {lvms-first} adds the supported unused devices to the LVM volume group. {lvms} adds the devices to the LVM volume group only if the following conditions are met:
+* The device path exists.
+* The device is supported by {lvms}. For information about unsupported devices, see _Devices not supported by {lvms}_ in the _Additional resources_ section.
```
There's an in-flight PR that has some actual guidance on this, but FWIW in this doc right now, this is the only spot where "Additional resources" is being italicized. For the rest of the assembly, it's been double-quotes for both the section name and `in the "Additional resources" section`.

sr1kar99 replied:
For now, added double-quotes to both the section name and "Additional resources" section.
Will update it later based on the finalized guidance in the PR to add in-module references.

adellape reviewed (commented):
Overall, looks good! I would like to hold merge, however, for a couple of user-replaceables using dashes. I'll keep the `merge-review-in-progress` label on it tho and I'll keep an eye out for an update. Thank you!

adellape commented on modules/lvms-creating-lvmcluster-using-rhacm.adoc:
```diff
@@ -0,0 +1,63 @@
++
+[source,terminal]
+----
+$ oc create -f <file_name> -n <ocp-cluster-namespace> <1>
```
On second thought, should it be explicitly saying `ocp-` here? This assembly AFAICT does not appear in the OSD or ROSA docs, but unless I'm missing a conditional related to this module in particular, the assembly does appear in the OKD docs [here](https://docs.okd.io/latest/storage/persistent_storage/persistent_storage_local/persistent-storage-using-lvms.html). And so it would look strange for the nearby `{product-title}`s to be resolving to "OKD", with this hard-coded `ocp-` in the replaceable:

<img width="585" alt="image" src="https://github.com/openshift/openshift-docs/assets/3442316/4e32f106-8b10-4609-8ee6-5b7b22981494">

sr1kar99 replied:
@adellape 
It makes sense to update it to "<cluster_namespace".

sr1kar99 commented on modules/lvms-creating-lvmcluster-using-rhacm.adoc:
```diff
@@ -0,0 +1,63 @@
+. Create a `ConfigurationPolicy` CR YAML file with the configuration to create an `LVMCluster` CR:
++
+.Example `ConfigurationPolicy` CR YAML file to create an `LVMCluster` CR
+[source, yaml]
```
Updated as per the suggestion.

sr1kar99 replied:
Updated as per the suggestion.

sr1kar99 commented on modules/lvms-deleting-lvmcluster-using-rhacm.adoc:
```diff
@@ -0,0 +1,192 @@
++
+[source,terminal]
+----
+$ oc delete -f <file_name> -n <ocp-cluster-namespace> <1>
```
Updated it to "<cluster_namespace>".

sr1kar99 commented:
> Overall, looks good! I would like to hold merge, however, for a couple of user-replaceables using dashes. I'll keep the `merge-review-in-progress` label on it tho and I'll keep an eye out for an update. Feel free to ping on Slack as well. Thank you!
> 
> Edit: Also added a question about using `ocp` in a replaceable without a conditional.

Hi @adellape 
Thank you for the detailed review.
Updated the content based on the suggestions.
Could you PTAL?
Thanks.

adellape commented:
/cherrypick enterprise-4.15
